src/scaffoldmaker/meshtypes/meshtype_3d_bone1.py

```diff
@@ -0,0 +1,461 @@
+                                                                      x[2]]
+        #
+        nodeIdentifier = max(1, getMaximumNodeIdentifier(nodes) + 1)
+        nodeIdentifier = hemisphere2.generateNodes(fm, coordinates, nodeIdentifier,
+                                                   rangeOfRequiredElements)
+
+        # replace the hemispherical nodes at interface with nodes on cylinder end
+        count = 0
+        hemisphereNodesToDelete2 = []
+        for n2 in range(elementsCountAcross[0] + 1):
+            for n1 in range(elementsCountAcross[1] + 1):
+                if hemisphere2.px[elementsCountAcross[2] // 2][n2][n1]:
+                    hemisphereNodesToDelete2.append(hemisphere2.nodeId[elementsCountAcross[2] // 2][n2][n1])
+                    hemisphere2.nodeId[elementsCountAcross[2] // 2][n2][n1] = nodesIdCylinderDistalEnd[count]
+                    count += 1
+
+        #         # rangeOfRequiredElements = [[0, elementsCountAcross[0]], [0, elementsCountAcross[1]], [0, 1]]
+        #         # nodeIdentifier = max(1, getMaximumNodeIdentifier(nodes) + 1)
+        #         # nodeIdentifier = hemisphere.generateNodes(fm, coordinates, nodeIdentifier,
+        #         #                                           rangeOfRequiredElements)
+        # #**********************************************************************************************************************
+        # generate hemisphere elements.
+        rangeOfRequiredElements = [[0, elementsCountAcross[0]], [0, elementsCountAcross[1]],
+                                   [int(0.5 * elementsCountAcross[2]),elementsCountAcross[2]]]
+        elementIdentifier = max(1, getMaximumElementIdentifier(mesh) + 1)
+        elementIdentifier = hemisphere2.generateElements(fm, coordinates, elementIdentifier,
+                                                         rangeOfRequiredElements)
+
+#         # AnnotationGroup.createMarkerNode(startNodeIdentifier=1, materialCoordinatesField: FieldFiniteElement = None, materialCoordinates = None, element = None, xi = [0.0, 0.0, 0.0])
+#
+#
+#         # markerTermNameBoneCoordinatesMap = {
+#         #     'tibial tuberosity': [-5.076472492200136e+01, -4.592226612078402e+01, -1.261953033704384e+03],
+#         # }
+#         # annotationGroups = []
+#         # nodes = fm.findNodesetByFieldDomainType(Field.DOMAIN_TYPE_NODES)
+#         # nodeIdentifier = max(1, getMaximumNodeIdentifier(nodes) + 1)
+#         # print(nodeIdentifier)
+#         # for termName, boneCoordinatesValues in markerTermNameBoneCoordinatesMap.items():
+#         #     annotationGroup = findOrCreateAnnotationGroupForTerm(annotationGroups, region, ('tibia point', 'FM:63'))
+#         #     annotationGroup.createMarkerNode(nodeIdentifier, coordinates, boneCoordinatesValues)
+#         #     nodeIdentifier += 1
+
+        # smoothing = DerivativeSmoothing(region, coordinates)
```

Comment:
Delete if not planning to use derivative smoothing

Reply:
Done